src/components/Menu/MenuItem.tsx

```diff
@@ -179,6 +297,55 @@ class MenuItem extends UIComponent<Extendable<MenuItemProps>, MenuItemState> {
 
     _.invoke(this.props, 'onFocus', e, this.props)
   }
+
+  private closeAllMenus = e => {
+    const { menu, inSubmenu } = this.props
+    const { menuOpen } = this.state
+    if (menu && menuOpen) {
+      this.setState({ menuOpen: false })
+      if (!inSubmenu) {
```

Comment:
for some cases we have this focus logic being run in the callback of the state change (which is correct), somewhere not. Lets make approach to that being consistent (or, if there is any reason why it is not, please, just let me know)

Reply:
Initially I had here check for which key was pressed, and I could not use the event in async manner there. That is not the case anymore, as we decided to use different handlers for different keys, see comment: https://github.com/stardust-ui/react/pull/539#discussion_r242924071 We can move this safely in the callback now.